src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -277,6 +277,10 @@ private static void handleXmlConfig(TestInputConfiguration.Builder testInputConf
                 setModuleName(moduleInputConfigBuilder, inputFilePath, moduleName);
                 setProperties(inputFilePath, module, moduleInputConfigBuilder);
                 testInputConfigBuilder.addChildModule(moduleInputConfigBuilder.build());
+
+                if ("Checker".equals(moduleName)) {
+                    handleXmlConfig(testInputConfigBuilder, inputFilePath, module.getChildren());
+                }
```

Comment:
if the current module is `Checker` then we're recursively calling the `handleXmlConfig()` method to set properties of it's children modules, this way the `testInputConfigBuilder` will contain Checker properties as well as Checker's children modules' properties.

Reply:
it is weird that `if ("Checker".equals(moduleName)) {` is below of `if ("TreeWalker".equals(moduleName)) {`
but it is better to fix it at new PR, here is new issue on this - https://github.com/checkstyle/checkstyle/issues/14653